packages/web-api/src/WebClient.ts

```diff
@@ -388,6 +393,143 @@ export class WebClient extends Methods {
+  Promise<Array<FilesGetUploadURLExternalResponse>> {
+    return Promise.all(fileUploads.map((upload: FileUploadV2Entry) => {
+      /* eslint-disable @typescript-eslint/consistent-type-assertions */
+      const options = {
+        filename: upload.filename,
+        length: upload.length,
+        alt_text: upload.alt_text,
+        snippet_type: upload.snippet_type,
+      } as FilesGetUploadURLExternalArguments;
+
+      return this.files.getUploadURLExternal(options);
+    }));
+  }
+
+  /**
+   * For each successful upload, completeUploadExternal endpoint,
+   * with a list of completed uploads - opt. channel id, initial comment, thread_ts
+   * @param fileUploads
+   * @returns
+   */
+  private async postCompletedFileUploads(fileUploads: FileUploadV2Entry[]):
+  Promise<Array<WebAPICallResult>> {  
+    const jobs = getAllFileUploadsToComplete(fileUploads);
+    const jobsValues = Object.values(jobs);
+
+    return Promise.all(
+      jobsValues.map((job: FilesCompleteUploadExternalArguments) => this.files.completeUploadExternal(job)),
+    );
+  }
+
+  private async getFileInfo(fileUploads: FileUploadV2Entry[]):
+  Promise<Array<WebAPICallResult>> {
+    return Promise.all(fileUploads.map((upload: FileUploadV2Entry) => this.files.info({ file: upload.file_id! })));
+  }
+
+  /**
+   * for each returned file upload URL, upload corresponding file
+   * @param fileUploads
+   * @returns
+   */
+  private async postFileUploadsToExternalURL(fileUploads: FileUploadV2Entry[], options: FilesUploadV2Arguments)
+    : Promise<Array<FilesGetUploadURLExternalResponse>> {
+    return Promise.all(fileUploads.map(async (upload: FileUploadV2Entry) => {
+      const { upload_url, file_id, filename, data } = upload;
+      // either file or content will be defined
+      const body = data;
+
+      // try to post to external url
+      if (upload_url) {
+        const headers: Record<string, string> = {};
+        if (options.token) headers.Authorization = `Bearer ${options.token}`;
+
+        const uploadRes = await this.makeRequest(upload_url, {
+          body,
+        }, headers);
+        if (uploadRes.status !== 200) {
+          return Promise.reject(Error(`Failed to upload file id:${file_id}, ${filename}`));
+        } 
+        const returnData = { ok: true, body: uploadRes.data } as WebAPICallResult;
+        return Promise.resolve(returnData);
+      } 
+      return Promise.reject(Error('No upload url found for file'));
+    }));
+  }
+
+  /**
+   * @param options All file uploads arguments
+   * @returns An array of file upload entries
+   */
+  private async getAllFileUploads(options: FilesUploadV2Arguments): Promise<FileUploadV2Entry[]> {
+    let fileUploads: FileUploadV2Entry[] = [];
+
+    // add single file data to uploads if file or content exists at the top level
+    if (options.file || options.content) {
+      const channels = options.channels ? options.channels.split(',') : [];
+      if (channels.length > 1) {
```

Comment:
In v2, sharing files in multiple channels at a time is unsupported. It may be technically possible to share the same set of files multiple files but even if it works, I don't think it's supported by design. The original methods (meaning the hidden methods without External suffix) for humans do not consider such a use case.  So, can you raise an exception in this scenario? See also: https://github.com/slackapi/python-slack-sdk/blob/v3.19.1/slack_sdk/web/client.py#L3034-L3040